tests/Commands/AppConfig/KeyValue/KeyValueLockCommandTests.cs

```diff
@@ -0,0 +1,160 @@
+        var result = JsonSerializer.Deserialize<KeyValueLockResult>(json);
+
+        Assert.NotNull(result);
+        Assert.Equal("my-key", result.Key);
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_LocksKeyValueWithLabel_WhenLabelProvided()
+    {
+        // Arrange
+        var command = new KeyValueLockCommand(_logger);
+        var args = command.GetCommand().Parse([
+            "--subscription", "sub123",
+            "--account-name", "account1",
+            "--key", "my-key",
+            "--label", "prod"
+        ]);
+        var context = new CommandContext(_serviceProvider);
+
+        // Act
+        var response = await command.ExecuteAsync(context, args);
+
+        // Assert
+        Assert.Equal(200, response.Status);
+        await _appConfigService.Received(1).LockKeyValue(
+            "account1",
+            "my-key",
+            "sub123", null,
+            Arg.Any<RetryPolicyOptions>(),
+            "prod");
+
+        var json = JsonSerializer.Serialize(response.Results);
+        var result = JsonSerializer.Deserialize<KeyValueLockResult>(json);
+
+        Assert.NotNull(result);
+        Assert.Equal("my-key", result.Key);
+        Assert.Equal("prod", result.Label);
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_Returns500_WhenServiceThrowsException()
+    {
+        // Arrange
+        _appConfigService.LockKeyValue(
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<RetryPolicyOptions>(),
+            Arg.Any<string>())
+            .Returns(Task.FromException<KeyValueSetting>(new Exception("Failed to lock key-value")));
+
+        var command = new KeyValueLockCommand(_logger);
+        var args = command.GetCommand().Parse([
+            "--subscription", "sub123",
+            "--account-name", "account1",
+            "--key", "my-key"
+        ]);
+        var context = new CommandContext(_serviceProvider);
+
+        // Act
+        var response = await command.ExecuteAsync(context, args);
+
+        // Assert
+        Assert.Equal(500, response.Status);
+        Assert.Contains("Failed to lock key-value", response.Message);
+    }
+    [Theory]
+    [InlineData("")]
+    [InlineData("--subscription sub123")]
+    [InlineData("--subscription sub123 --account-name account1")]
+    public async Task ExecuteAsync_Returns400_WhenRequiredParametersAreMissing(string args)
+    {
+        // Arrange
+        var command = new KeyValueLockCommand(_logger);
+        var parsedArgs = command.GetCommand().Parse(args.Split(' ', StringSplitOptions.RemoveEmptyEntries));
+        var context = new CommandContext(_serviceProvider);
+
+        // Act
+        var response = await command.ExecuteAsync(context, parsedArgs);
+
+        // Assert
+        Assert.Equal(400, response.Status);
+        Assert.Contains("required", response.Message.ToLower());
+    }
+
+    private sealed class KeyValueLockResult
```

Comment:
Maybe we have a /models folder in the tests project if they don't go into the main project?